dashboard/modules/job/job_head.py

```diff
@@ -57,32 +83,81 @@ async def get_package(self,
     async def upload_package(self, req: aiohttp.web.Request):
         package_uri = req.query["package_uri"]
         logger.info(f"Uploading package {package_uri} to the GCS.")
-        upload_package_to_gcs(package_uri, await req.read())
+        try:
+            upload_package_to_gcs(package_uri, await req.read())
+        except Exception:
+            return aiohttp.web.Response(
+                reason=traceback.format_exc().encode("utf-8"),
+                status=aiohttp.web.HTTPInternalServerError.status_code)
 
-        return aiohttp.web.Response()
+        return aiohttp.web.Response(
+            text={"success": True},
```

Comment:
do we ever set `success = False`? otherwise probably don't need this?

Reply:
we don't really use it, status code is only thing we need